src/Renderer/Inline/TextRenderer.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file was originally part of the league/commonmark package.
+ *
+ * (c) Colin O'Dell <colinodell@gmail.com>
+ *
+ * Original code based on the CommonMark JS reference parser (https://bitly.com/commonmark-js)
+ *  - (c) John MacFarlane
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace UnicornFail\Emoji\Renderer\Inline;
+
+use UnicornFail\Emoji\Node\Inline\Text;
+use UnicornFail\Emoji\Node\Node;
+use UnicornFail\Emoji\Renderer\ChildNodeRendererInterface;
+use UnicornFail\Emoji\Renderer\NodeRendererInterface;
+
+final class TextRenderer implements NodeRendererInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function render(Node $node, ChildNodeRendererInterface $childRenderer)
+    {
+        if (! ($node instanceof Text)) {
+            throw new \InvalidArgumentException('Incompatible node type: ' . \get_class($node));
```

Comment:
Will there only be a finite number of node types?  In other words, do we ever plan to support more than just emojis, shortcodes, emoticons, and whatever else you might have implemented?

If the types are finite and new types cannot be added by users then we might be able to simplify the rendering process in a more type-strict way.  I don't have any specific ideas or suggestions at the moment - just curious right now :)

Reply:
> Will there only be a finite number of node types?

At this moment in time, I do not know. I really don't want to box the project in though and would like to leave this open to interpretation. My goal is to get away from switch/if statements based on the lexer "type" and would rather let a node indicate whether it matches. I'm still not 💯 how I want to do this.

The primary goal of this though was to show the logical workflow as to what would be considered a "node". If some aspects of commonmark were extracted into a simpler library that both of these projects could share (i.e. `league/ast` ) or at the very least some sort of rudimentary interface that could be shared, then these nodes could be passed directly to commonmark as is (there wouldn't be any need to "transform them into commonmark nodes").

Reply:
With extensions, yes, this can be any kind of Node (i.e. images)